src/computability/DFA.lean

```diff
@@ -43,4 +44,129 @@ def eval := M.eval_from M.start
 def accepts : language α :=
 λ x, M.eval x ∈ M.accept
 
+lemma mem_accepts (x : list α) : x ∈ M.accepts ↔ M.eval_from M.start x ∈ M.accept := by refl
+
+/-- `M.state_path s x` is the states the DFA passes through when evalulating `M.eval_from s x` -/
+def state_path : σ → list α → list σ
+| start [] := [start]
+| start (a :: x) := start :: state_path (M.step start a) x
+
+@[simp] lemma length_state_path (start : σ) (x : list α) :
+  (M.state_path start x).length = x.length + 1 :=
+begin
+  induction x generalizing start,
+  refl,
+  tauto
+end
```

Comment:
This should be a lemma about `list.scanl` I think

Reply:
```suggestion
@[simp] lemma length_state_path (start : σ) (x : list α) :
  (M.state_path start x).length = x.length + 1 :=
list.length_scanl start x
```

Reply:
With dot notation:
```suggestion
x.length_scanl start
```